File: src/main/java/leets/leenk/domain/birthday/domain/service/BirthdayGetService.java
package leets.leenk.domain.birthday.domain.service;

import leets.leenk.domain.user.domain.entity.User;
import leets.leenk.domain.user.domain.repository.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.time.LocalDate;
import java.util.List;

@Service
@RequiredArgsConstructor
public class BirthdayGetService {
    private final UserRepository userRepository;

    public List<User> findTodayBirthdayUsers(LocalDate today) {
        return userRepository.findAllUsersInBirthday(today.getMonthValue(), today.getDayOfMonth());
    }

    public List<User> findUpcomingBirthdayUsers(LocalDate today, int days) {
        return userRepository.findUpcomingBirthdays(today, days);
    }
}

File: src/main/kotlin/leets/leenk/domain/birthday/domain/service/BirthdayGetService.kt
package leets.leenk.domain.birthday.domain.service

import leets.leenk.domain.user.domain.entity.User
import leets.leenk.domain.user.domain.repository.UserRepository
import org.springframework.stereotype.Service
import java.time.LocalDate

@Service
class BirthdayGetService(
    private val userRepository: UserRepository,
) {
    fun findTodayBirthdayUsers(today: LocalDate): List<User> =
        userRepository.findAllUsersInBirthday(today.monthValue, today.dayOfMonth)

    fun findUpcomingBirthdayUsers(
        today: LocalDate,
        days: Int,
    ): List<User> = userRepository.findUpcomingBirthdays(today, days)
}
